nemo/collections/nlp/models/intent_slot_classification_model.py

```diff
@@ -0,0 +1,264 @@
+        """
+
+        # init superclass
+        super().__init__(cfg=cfg, trainer=trainer)
+
+        data_desc = IntentSlotDataDesc(data_dir=cfg.data_dir, modes=["train", "test", "dev"])
+
+        # initialize Tokenizer
+        self.tokenizer = get_tokenizer(
+            tokenizer_name=cfg.language_model.tokenizer,
+            pretrained_model_name=cfg.language_model.pretrained_model_name,
+            vocab_file=cfg.language_model.vocab_file,
+            tokenizer_model=cfg.language_model.tokenizer_model,
+            do_lower_case=cfg.language_model.do_lower_case,
+        )
+
+        # initialize Bert model
+        self.bert_model = get_pretrained_lm_model(
+            pretrained_model_name=cfg.language_model.pretrained_model_name,
+            config_file=cfg.language_model.bert_config,
+            checkpoint_file=cfg.language_model.bert_checkpoint,
+        )
+
+        self.data_dir = cfg.data_dir
+        self.max_seq_length = cfg.language_model.max_seq_length
+        self.hidden_size = self.bert_model.config.hidden_size
+
+        self.classifier = SequenceTokenClassifier(
+            hidden_size=self.hidden_size,
+            num_intents=data_desc.num_intents,
+            num_slots=data_desc.num_slots,
+            dropout=cfg.head.fc_dropout,
+            num_layers=cfg.head.num_output_layers,
+            log_softmax=False,
+        )
+
+        # define losses
+        if cfg.class_balancing == 'weighted_loss':
+            # You may need to increase the number of epochs for convergence when using weighted_loss
+            self.intent_loss = CrossEntropyLoss(logits_ndim=2, weight=data_desc.intent_weights)
+            self.slot_loss = CrossEntropyLoss(logits_ndim=3, weight=data_desc.slot_weights)
+        else:
+            self.intent_loss = CrossEntropyLoss(logits_ndim=2)
+            self.slot_loss = CrossEntropyLoss(logits_ndim=3)
+
+        self.total_loss = AggregatorLoss(num_inputs=2, weights=[cfg.intent_loss_weight, 1.0 - cfg.intent_loss_weight])
+
+        # setup to track metrics
+        self.intents_classification_report = ClassificationReport(
+            self.data_desc.num_intents, self.data_desc.intents_label_ids
+        )
+        self.slots_classification_report = ClassificationReport(
+            self.data_desc.num_slots, self.data_desc.slots_label_ids
+        )
+
+        # Optimizer setup needs to happen after all model weights are ready
+        self.setup_optimization(cfg.optim)
+
+    @typecheck()
+    def forward(self, input_ids, token_type_ids, attention_mask):
+        """
+        No special modification required for Lightning, define it as you normally would
+        in the `nn.Module` in vanilla PyTorch.
+        """
+        hidden_states = self.bert_model(
+            input_ids=input_ids, token_type_ids=token_type_ids, attention_mask=attention_mask
+        )
+        intent_logits, slot_logits = self.classifier(hidden_states=hidden_states)
```

Comment:
This code is not going to work. You should have two classification losses here.

Reply:
We discussed it, I implemented 2 heads in the classifier.